crypto/evp/e_aes.c

```diff
@@ -2127,6 +2127,10 @@ static int aes_ccm_cipher(EVP_CIPHER_CTX *ctx, unsigned char *out,
     if (cctx->tls_aad_len >= 0)
         return aes_ccm_tls_cipher(ctx, out, in, len);
 
+    /* EVP_*Final() doesn't return any data */
+    if (in == NULL && out != NULL)
+        return 0;
```

Comment:
Due to a bug in the CCM code EVP_*Final() used to always return an error if any data had been processed: that was because EVP_*Update() in CCM mode resets the iv_set, tag_set and len_set fields in (as CCM mode can only call EVP_*Update once).